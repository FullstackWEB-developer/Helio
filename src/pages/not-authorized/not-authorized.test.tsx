import i18n from '../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {fireEvent, render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import NotAuthorized from './not-authorized';
jest.setTimeout(30000);
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("NotAuthorized tests", () => {
    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders NotAuthorized correctly", async () => {
        const {asFragment} = render(<TestWrapper mockState={mockState}>
                <NotAuthorized/>          
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });
})