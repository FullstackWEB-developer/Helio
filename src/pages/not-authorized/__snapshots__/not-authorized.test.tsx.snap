// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotAuthorized tests renders NotAuthorized correctly 1`] = `<DocumentFragment />`;
